transport-native-io_uring/src/main/c/netty_io_uring_native.c

```diff
@@ -276,28 +244,22 @@ static void netty_io_uring_ring_buffer_exit(JNIEnv *env, jclass class, jobject r
 	close(submissionQueueRingFd);
 }
 
-static int nettyBlockingSocket(int domain, int type, int protocol) {
-    return socket(domain, type, protocol);
-}
-
 static jobject netty_io_uring_setup(JNIEnv *env, jclass class1, jint entries, jobject submitCallback) {
     struct io_uring_params p;
     memset(&p, 0, sizeof(p));
 
     int ring_fd = sys_io_uring_setup((int)entries, &p);
 
-    //Todo
     if (ring_fd < 0) {
-      printf("RingFd error: %d\n", ring_fd);
-      //throw Exception
+     netty_unix_errors_throwRuntimeExceptionErrorNo(env, "failed to create io_uring ring fd ", errno);
       return NULL;
     }
     struct io_uring io_uring_ring;
-    //Todo memset instead
-    io_uring_ring.flags = 0;
-    io_uring_ring.sq.sqe_tail = 0;
-    io_uring_ring.sq.sqe_head = 0;
-    setup_io_uring(ring_fd, &io_uring_ring, &p);
+    int ret = setup_io_uring(ring_fd, &io_uring_ring, &p);
+
+    if (ret != 0) {
+        netty_unix_errors_throwRuntimeExceptionErrorNo(env, "failed to mmap io_uring ring buffer", ret);
```

Comment:
shouldn't that better include the `errno` ?

Reply:
yeah sure, however we already return a errno in `setup_io_uring`, I thought it would be more readable

Reply:
ah ok .... missed that. Then its fine :)